### DISCUSSION TITLE
The "pathname" param on "onRenderBody" is still broken in the dev build.

### DISCUSSION THREAD

**Discussion by Johnrobmiller** [gatsbyjs/gatsby#32858] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

The "pathname" param on "onRenderBody" is broken in the dev build.  This has been broken for over a year now, and other people have reported it.  If you guys don't intent to fix this, then please remove the mention of "pathname" from the documentation. That way, you would save people from headaches since the inclusion of "pathname" in the docs is misleading people.  Plus, I can't imagine who could possibly use this feature without being able to test it locally before publishing.

Also, in general, I would stop publishing half-finished features like they're ready for use.  This cannot possibly be good design practice.

### Reproduction Link

.

### Steps to Reproduce

.

### Expected Result

.

### Actual Result

.

### Environment

```shell
.
```

**Comment by LekoArts** [comment]
Hi!

While I can understand that one sometimes want to vent I don't think it's a good idea to do that to maintainers of any open source project. We're (mostly) always happy to help but have limited resources, sometimes miss a notification or get pulled into thousands of different directions.

Users can help by writing proper issues (following the template), keeping a friendly tone, and having the best intentions to help solve the issue. You could have linked the issues you meant so that I didn't have to spend time searching for myself for example.

What you're referring to is not "broken", it's a limitation of how we currently construct the `gatsby develop` environment. So while the DX is not the best, it's working as expected.

However, you can use https://github.com/gatsbyjs/gatsby/discussions/28138 to get your desired result to get the information during `gatsby develop`, too.

**Reply by Johnrobmiller** [reply]
> So while the DX is not the best, it's working as expected.

For whom exactly is this working as expected?  You guys, the deverlops of Gatsby, or us, the users of Gatsby?  I think that you are thinking about this in a backwards way.
